packages/material-ui/src/TableRow/TableRow.js

```diff
@@ -1,65 +1,91 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import Tablelvl2Context from '../Table/Tablelvl2Context';
 import { alpha } from '../styles/colorManipulator';
 
-export const styles = (theme) => ({
+import useThemeProps from '../styles/useThemeProps';
+import experimentalStyled from '../styles/experimentalStyled';
+import { getTableRowUtilityClass } from './tableRowClasses';
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(styleProps.head && styles.head),
+    ...(styleProps.footer && styles.footer),
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, selected, hover, head, footer } = styleProps;
+
+  const slots = {
+    root: ['root', selected && 'selected', hover && 'hover', head && 'head', footer && 'footer'],
+  };
+
+  return composeClasses(slots, getTableRowUtilityClass, classes);
+};
+
+const TableRowRoot = experimentalStyled(
+  'tr',
+  {},
+  {
+    name: 'MuiTableRow',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ theme }) => ({
   /* Styles applied to the root element. */
-  root: {
-    color: 'inherit',
-    display: 'table-row',
-    verticalAlign: 'middle',
-    // We disable the focus ring for mouse, touch and keyboard users.
-    outline: 0,
-    '&$hover:hover': {
-      backgroundColor: theme.palette.action.hover,
-    },
-    '&$selected, &$selected:hover': {
-      backgroundColor: alpha(theme.palette.secondary.main, theme.palette.action.selectedOpacity),
-    },
+  color: 'inherit',
+  display: 'table-row',
+  verticalAlign: 'middle',
+  // We disable the focus ring for mouse, touch and keyboard users.
+  outline: 0,
+  '&.Mui-hover:hover': {
```

Comment:
This is not s valid MUI class, please use `tableRowClasses.hover`

Reply:
I thought you were off?

Reply:
I will pop out from time to time :)